xjc-maven-plugin/src/main/java/com/github/davidmoten/xjc/XjcMojo.java

```diff
@@ -246,10 +295,10 @@ private static String string(Artifact a) {
     private void ensureDestinationDirectoryExists() {
         for (int i = 0; i < arguments.size(); i++) {
             if (arguments.get(i).trim().equals("-d") && i < arguments.size() - 1) {
-                File dir = new File(arguments.get(i + 1));
-                if (!dir.exists()) {
-                    getLog().info("destination directory (-d option) specified and does not exist, creating: " + dir);
-                    dir.mkdirs();
+                outputDir = new File(arguments.get(i + 1));
```

Comment:
we don't need a field here, just return a File from the method. The -d argument does not have to be specified. If not specified the default is the current directory which this method should return if no -d argument specified. That way we won't pass null to `buildContext.refresh`.

Reply:
Ok, will fix it